File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/transactionpropagation/TransactionSentTraceEvent.java
package org.palladiosimulator.blockchainsystems.core.transactionpropagation;

import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetworkEndpoint;

public record TransactionSentTraceEvent(
        long occurrenceTime,
        Transaction sentTransaction,
        P2PNetworkEndpoint receivingEndpoint
) implements TraceEvent {

    public static final String EVENT_TYPE = "TransactionSentTraceEvent";

    @Override
    public long getOccurrenceTime() {
        return occurrenceTime;
    }

    public Transaction getSentTransaction() {
        return sentTransaction;
    }

    public P2PNetworkEndpoint getReceivingNetworkEndpoint() {
        return receivingEndpoint;
    }

    @Override
    public String getEventType() {
        return EVENT_TYPE;
    }

    @Override
    public void formatDetails(StringBuilder stringBuilder) {
        stringBuilder.append("{ txId=");
        stringBuilder.append(sentTransaction.getTxId());
        stringBuilder.append("; sentToNetworkEndpoint=");
        stringBuilder.append(receivingEndpoint.getEndpointId());
        stringBuilder.append(" }");
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/transactionpropagation/TransactionSentTraceEvent.kt
package org.palladiosimulator.blockchainsystems.core.transactionpropagation

import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent
import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetworkEndpoint
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction

/**
 * @author Davis Riedel
 */
@JvmRecord
data class TransactionSentTraceEvent(
  private val occurrenceTime: Long,
  val sentTransaction: Transaction?,
  val receivingNetworkEndpoint: P2PNetworkEndpoint?
) : TraceEvent {
  companion object {
    const val EVENT_TYPE: String = "TransactionSentTraceEvent"
  }

  override fun getOccurrenceTime(): Long = occurrenceTime
  override fun getEventType(): String = EVENT_TYPE

  override fun formatDetails(stringBuilder: StringBuilder) {
    stringBuilder.append("{ txId=")
    stringBuilder.append(sentTransaction!!.getTxId())
    stringBuilder.append("; sentToNetworkEndpoint=")
    stringBuilder.append(receivingNetworkEndpoint!!.getEndpointId())
    stringBuilder.append(" }")
  }
}